### ISSUE TITLE
Failed migration of SQLite from 0.0.16 to 0.0.17

### ISSUE THREAD

**Issue by IanMayo** [debrief/pepys-import#476]
While I was able to migrate a PG database from 0.0.16 to 0.0.17, trying to do it on SQLite failed:
```
(pepys-admin) 5
Your database schema is going to be updated. Are you sure to continue? (y/N) y
Alembic migration command running, see output below.
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade b03fb305981b -> 4fb1c8780273, Add unique constraint for Platform
Exception details: (sqlite3.IntegrityError) FOREIGN KEY constraint failed
[SQL: 
DROP TABLE "Platforms"]
(Background on this error at: http://sqlalche.me/e/gkpj)

ERROR: Alembic error when migrating the database!
----------------------------------
```

This is largely FYI - I'm going to be doing PG migration, then taking a fresh snapshot.

The above may help us understand SQLite migrations, but I think @BarisSari  has discussed this issue before.

**Comment by robintw** [comment]
I _think_ this is a problem that is occurring because when doing this migration for SQLite, it has to create a new table, copy things across, drop the old table, and then rename the copied table. This will break foreign key constraints when the old table is deleted.

I'm sure there must be a way around this - I guess we could turn off foreign key constraint checking (you have to explicitly turn it on for SQLite - so we could just remove that bit for when alembic runs), but I'm wondering if that might cause us more problems...

**Comment by IanMayo** [comment]
Thanks @robintw - I'm hoping we can mature the schema so that changes like this occur less and less frequently.

We also have the "model" where there aren't persistent SQLite databases. They're just for deployed working or development.  So, wiping them isn't an issue.

Hmm, _maybe_ we need to ensure they're at the same version when we merge. That would require migration of the SQLite snapshot.

**Comment by robintw** [comment]
Yes, it would probably be a good idea to make sure that the databases are at the same version when we merge..._but_ I think that'll be fairly difficult, as we'll be merging a SQLite database into a Postgres database, and the versions don't necessarily completely match up. That is, we can have different numbers of migrations for SQLite and Postgres, and their GUIDs don't match anyway. I think it'll be fairly difficult to work out what counts as 'matching'....but I agree we'll have problems if they don't match (assuming it is a substantive change like adding/removing/renaming a column...if it's just adding an extra constraint or something then it shouldn't cause many problems).

We don't need to migrate the SQLite snapshot at the moment, as the snapshot will be coming from the most up-to-date schema when it is exported from the main database today. We will have to think about what we do if we do any substantive updates to the schema on the master database while the slave is away at sea - I'll talk to @BarisSari about that. As you said though, as the schema gets more mature, we should have less of these migrations to worry about.

**Comment by BarisSari** [comment]
Hi @IanMayo and @robintw . We already turned off foreign key constraint enforcement on Alembic. Otherwise, it breaks referential integrity: 

https://alembic.sqlalchemy.org/en/latest/batch.html#dealing-with-referencing-foreign-keys

I'll think about how we can understand that both databases are on the same version. Maybe we might add extra fields to `latest_revisions.json` and check it.

**Comment by robintw** [comment]
This was fixed by #502. I'll create a separate issue for the question about merging databases with different versions.
